packages/@aws-cdk/aws-codepipeline-api/lib/action.ts

```diff
@@ -151,6 +151,15 @@ export interface CommonActionConstructProps {
    * The Pipeline Stage to add this Action to.
    */
   stage: IStage;
+
+  /**
+   * The service role that is assumed during execution of action.
+   * This role is not mandatory, however more advanced configuration
+   * may require specifying it.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codepipeline-pipeline-stages-actions.html
+   */
+  actionRole?: iam.IRole;
```

Comment:
Can we just call this `role`?

Reply:
`PipelineCloudFormationDeployActionProps` uses `role` attribute to describe role assumed by Cloud Formation. The ref. arch. actually suggest usage of both roles for cross-account deployments.

If we would rename this attribute, one on *CloudFormation* should be renamed too, which would occur in a backward incompatible change.

Reply:
I wonder whether it's actually worth it to change the attribute in `PipelineCloudFormationDeployActionProps` to something like `cloudFormationRole`, and name this new one `role`. The reason is that this one is much more generic than the CFN Actions one, so the name `role` there does not seem entirely appropriate.

@eladb what do you think?

Reply:
@skinny85 not a bad idea. Maybe something like `deploymentRole`? (it is a deploy action)

Reply:
`deploymentRole` sounds a little ambiguous to me, but sure. @rsmogura can you make this change? Thanks!

Reply:
I think the name like `deploymentRole` would look ok (refarch for cross account deployments uses similar term *deployer*), and definitely it should be changed.

I would like to point out that if we will call the `actionRole` a `role` it can be backward incompatible and hardly captured by TypeScript transpiler - if someone in existing code already have set a `role` for CF action after update behavior of this property will change. So I just would like to be sure if it's ok.

I think we should as well consider changing and moving `addToRolePolicy` to `Action`, and changing this method on CF action to something like `addToDeploymentRolePolicy`?

Reply:
I renamed CF `role` to `deploymentRole` and `PipelineDeployStackAction` in `app-delivery`, changed `addToRolePolicy` method names to `addToDeploymentRolePolicy`.

I do not update action's service role permissions when added to pipeline. I think as feature is for advanced use-cases role or artifacts can come from other stack(s).

I tested this change with simple pipeline and setting action's role on source action, requires adding **s3** and **codecommit** actions to be granted by developer.

Reply:
@skinny85  - I don't think of `deploymentRole` as so ambiguous in that context... Thats the way at least I think of it there... And it maps nicely with the user intention... So that's good. As I mentioned in a separate note, `deployRole` might be another option (I have no strong feelings either way).